examples/librbd/Makefile

```diff
@@ -1,6 +1,6 @@
 
 CXX?=g++
-CXX_FLAGS?=-std=c++11 -Wno-unused-parameter -Wall -Wextra -Werror -g
+CXX_FLAGS?=-Wno-unused-parameter -Wall -Wextra -Werror -g
```

Comment:
Hrm, why drop `-std` option instead of bumping it to C++17 like https://github.com/ceph/ceph/pull/44738 attempted to do?  The version of the standard that is picked by default depends on the version of the compiler installed and the error message in case the explicitly specified version isn't supported by the compiler is much better than in case of compiling an unsupported syntax/construct.

Reply:
The commit just (blindly) follows be838652fae1ee4683b62785192a58992a76d1fe.

> The version of the standard that is picked by default depends on the version of the compiler installed 

I perceive the lack of `-std=...` as an avoidance of making any claim / hint on the language. At the moment I'm not sure this it's best option but whatever we choose, we should be consistent

Reply:
> The commit just (blindly) follows [be83865](https://github.com/ceph/ceph/commit/be838652fae1ee4683b62785192a58992a76d1fe).

Note that this commit actually passes `-std=c++20` for Ubuntu despite it being dropped from the Makefile ;)

I think it's actually redundant -- the problem in https://tracker.ceph.com/issues/57163 was likely not the `-std` switch but just the old compiler.  Simply upgrading the compiler and leaving `-std=c++11` in place would have worked.

I'm not expressing objection to this PR, but rather just pointing out that https://github.com/ceph/ceph/commit/be838652fae1ee4683b62785192a58992a76d1fe is inconsistent within itself.